src/geometry/partitions/distributedptscotchunstructuredpartition.h

```diff
@@ -0,0 +1,291 @@
+        end = (float)(rank + 1) / numPartitions * cells;
+    }
+
+    Region<DIM> getRegion(const std::size_t node) const override
+    {
+        return regions.at(node);
+    }
+
+private:
+    Coord<DIM> origin;
+    Coord<DIM> dimensions;
+    SCOTCH_Num localCells = 0;
+    size_t start = 0, end = 0;
+    size_t numPartitions = 0;
+    std::vector<Region<DIM> > regions;
+
+    void buildRegions()
+    {
+        std::vector<SCOTCH_Num> indices(localCells);
+
+        {
+            auto start = std::chrono::system_clock::now();
+            initIndices(indices);
+            auto duration = std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::system_clock::now() - start);
+            std::cout << "partitioning duration: " << duration.count() << std::endl;
+        }
+
+        regions.resize(numPartitions);
+
+        {
+            auto start = std::chrono::system_clock::now();
+            createRegions(indices);
+            auto duration = std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::system_clock::now() - start);
+            std::cout << "region creation duration: " << duration.count() << std::endl;
+
+        }
+    }
+
+    void initIndices(std::vector<SCOTCH_Num> &indices)
+    {
+        std::cout << "partitioning ..." << std::endl;
+
+        // create 2D grid
+        SCOTCH_Dgraph graph;
+        int error = SCOTCH_dgraphInit(&graph, MPILayer().communicator());
+
+        SCOTCH_Num numEdges = 0;
+
+        for (auto &p : this->adjacency)
+        {
+            numEdges += p.second.size();
+        }
+
+        SCOTCH_Num *verttabGra = new SCOTCH_Num[localCells + 1];
+        SCOTCH_Num *edgetabGra = new SCOTCH_Num[numEdges];
+
+        int currentEdge = 0;
+        for (int i = 0; i < localCells; ++i)
+        {
+            verttabGra[i] = currentEdge;
+
+            for (int other : this->adjacency[start + i])
+            {
+                edgetabGra[currentEdge++] = other;
+            }
+        }
+
+        numEdges = currentEdge;
+
+        verttabGra[localCells] = currentEdge;
+
+        int ierr = 0;
+        error = SCOTCH_dgraphBuild(&graph,
+                0,              // c++ starts counting at 0
+                localCells,     // number of local vertices
+                localCells,     // number of maximum local vertices to be created
+                verttabGra,     // vertex data
+                nullptr,        // these are some ghost zone
+                nullptr,        // and other hints that are optional, so -> nullptr
+                nullptr,        //
+                numEdges,       // number of local edges
+                numEdges,       // number of maximum local edges to be created
+                edgetabGra,     // edge data
+                nullptr,        // more optional data/hints
+                nullptr);       //
+        if (error) std::cout << "SCOTCH_graphBuild error: " << error << ", ierr: " << ierr << std::endl;
```

Comment:
Coding style:
- curly braces for all if-constructs
- use LOG() instead of std::cout
- move if-body to separate line